runc.go

```diff
@@ -174,13 +182,17 @@ func (r *Runc) Create(context context.Context, id, bundle string, opts *CreateOp
 	cmd.ExtraFiles = opts.ExtraFiles
 
 	if cmd.Stdout == nil && cmd.Stderr == nil {
-		data, err := cmdOutput(cmd, true, nil)
+		data, err := r.cmdOutput(cmd, true, nil)
 		defer putBuf(data)
 		if err != nil {
 			return fmt.Errorf("%s: %s", err, data.String())
 		}
 		return nil
 	}
+	if r.PdeathSignal != 0 {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+	}
 	ec, err := Monitor.Start(cmd)
```

Comment:
`Monitor.Start` also creates goroutines, using a separate goroutine to Wait for the child process.  I think it would make sense to do the following:

* Push the thread locking logic into `Monitor.Start` (or perhaps make a wrapper `Monitor.StartLocked`)
* Move `cmd.Start` into the same inner goroutine as where `cmd.Wait` is called, and then lock _that_ one

This has the advantage of (1) minimizing the scope of the locked thread (constrained to just around the process-management bits) and (2) keeps the locking logic close to where the process-management logic lives.  Otherwise there's a risk that a future author and reviewer may miss the importance of thread locking and could change the behavior of `Monitor.Start`.

Here's what that could look like:

```go
// StartLocked is like Start, but locks the goroutine used to start the process to
// the OS thread for use-cases where the parent thread matters to the child process
// (for example: when Pdeathsig is set).
func (m *defaultMonitor) StartLocked(c *exec.Cmd, locked bool) (chan Exit, error) {
	started := make(chan error)
	ec := make(chan Exit, 1)
	go func() {
		if locked {
			runtime.LockOSThread()
			defer runtime.UnlockOSThread()
		}
		if err := c.Start(); err != nil {
			started <- err
			return
		}
		close(started)
		var status int
		if err := c.Wait(); err != nil {
			status = 255
			if exitErr, ok := err.(*exec.ExitError); ok {
				if ws, ok := exitErr.Sys().(syscall.WaitStatus); ok {
					status = ws.ExitStatus()
				}
			}
		}
		ec <- Exit{
			Timestamp: time.Now(),
			Pid:       c.Process.Pid,
			Status:    status,
		}
		close(ec)
	}()
	if err := <-started; err != nil {
		return nil, err
	}
	return ec, nil
}
```

Reply:
Thanks for the suggestion. I've added a `StartLocked` method similar to this, which unconditionally uses `LockOSThread`, and a wrapper in `Runc` which uses either `Start` or `StartLocked`.